tests/integration/utils/test_fork_utils.py

```diff
@@ -0,0 +1,54 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from pathlib import Path
+
+import pytest
+from craft_parts.utils import fork_utils
+
+
+@pytest.fixture
+def case_dir() -> Path:
+    return Path(__file__).parent / "test_fork_utils"
+
+
+def test_simple_script(case_dir, capfd) -> None:
+    fork_utils.run(["/bin/bash", case_dir / "simple.sh"])
+    assert capfd.readouterr().out == "foo\n"
+
+
+def test_complex_script(case_dir, capfd) -> None:
+    result = fork_utils.run(["/bin/bash", case_dir / "complex.sh"])
+
+    out, err = capfd.readouterr()
+    out_n = [int(s) for s in out.split()]
```

Comment:
They should be 100 and 300, respectively - hardcoded from that `complex.sh` script

Reply:
what I meant was - please check that the lists have those expected lengths, because it kind of looks like the test might still pass if `out` and `err` are empty